Exclude vertical length offset as 1st component in 2 component positi…

skynavga opened the pull request:
…on (#711).

Closes #711.
Closes #809.

nigelmegitt commented:
Build available at https://rawgit.com/w3c/ttml2/issue-0711-position-build/index.html

skynavga commented on spec/ttml2.xml:
```diff
@@ -15091,7 +15091,9 @@ edge-keyword-v
-<p>a horizontal offset (<code>offset-position-h</code>) and a vertical offset (<code>offset-position-v</code>), in any order</p>
+<p>a horizontal offset (<code>offset-position-h</code>) and a vertical offset (<code>offset-position-v</code>), in any order; except
+that a sequence of two components consisting of a <code><loc href="#style-value-length">&lt;length&gt;</loc></code> expression
+followed by a <code>edge-keyword-h</code> token is not allowed</p>
```
It is allowed, since it preserves the semantic of _first <length> is a horizontal offset_.

nigelmegitt replied:
That's not the semantic we're targeting though. I understood from our discussion yesterday that we are trying to match the constraints that CSS introduced, which I take to be the latest CSS snapshot, which is [CSS Backgrounds Level 3](https://www.w3.org/TR/css-backgrounds-3/#the-background-position) which states under `bg-position`:

> If two values are given, a `<length-percentage>` as the first value represents the horizontal position (or offset) and a `<length-percentage>` as the second value represents the vertical position (or offset).

and:

> Note that a pair of keywords can be reordered while a combination of keyword and length or percentage cannot. So center left is valid while 50% left is not. 

Therefore `top 50%` is not allowed, i.e. `<length>` expression preceded by a `edge-keyword-v` token is _not_ allowed.

palemieux replied:
```
  | offset-position-h <lwsp> offset-position-v    // two component values
  | offset-position-v <lwsp> offset-position-h    // two component values
```

should be replaced with

```
  | position-keyword-h <lwsp> position-keyword-v
  | position-keyword-v <lwsp> position-keyword-h
  | position-keyword-h <lwsp> <length>
  | <length> <lwsp> position-keyword-v
  | <length> <lwsp> <length>
```

... or equivalent.

skynavga commented:
@palemieux please re-review after commit https://github.com/w3c/ttml2/pull/769/commits/2ffd951340c31b876aaebb3892bb2f511e791b9d

nigelmegitt commented on spec/ttml2.xml:
```diff
@@ -15037,14 +15037,17 @@ If no <emph>units</emph> component is specified, then Hertz is implied.</p>
 &lt;position&gt;
   : offset-position-h                             // single component value
   | offset-position-v                             // single component value
```
I've raised #809 as I see from reviewing this pull request that line 15039 is problematic - probably the easiest fix is to remove it in this pull request, which would partially resolve #809 too (would also need to fix the one component mapping to two components, in two places later on).

skynavga replied:
Addressed in https://github.com/w3c/ttml2/pull/769/commits/b6eb634ef00d6c38880aa8cc2ce5a209a753f24a. Also addressed #809.

skynavga commented:
@nigelmegitt addressed your comment above, and made further simplifications, one being to collapse four of rules proposed by @palemieux into one exactly equivalent rule

skynavga commented:
@palemieux ping
